onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -62,6 +62,65 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
     : rand_engine_(rd()), input_names_(m.GetInputCount()), input_names_str_(m.GetInputCount()), input_length_(m.GetInputCount()) {
   Ort::SessionOptions session_options;
 
+  bool is_plugin_ep_available = false;
+
+  // Add devices created from plugin EP
+  if (!performance_test_config.registered_plugin_eps.empty()) {
+    std::vector<Ort::ConstEpDevice> ep_devices = env.GetEpDevices();
+    std::vector<Ort::ConstEpDevice> added_ep_devices;
+    std::unordered_set<int> added_ep_device_index_set;
+
+    // Select devices by provided device index
+    if (!performance_test_config.selected_devices.empty()) {
+      std::vector<int> device_list;
+      device_list.reserve(performance_test_config.selected_devices.size());
+      ParseDeviceList(performance_test_config.selected_devices, device_list);
+      for (auto index : device_list) {
+        if (static_cast<size_t>(index) > (ep_devices.size() - 1)) {
+          fprintf(stderr, "%s", "The device index provided is not correct. Will skip this device id.");
```

Comment:
should it skip the rest of the iteration? otherwise, there will be an out of bounds access at `Ort::ConstEpDevice& device = ep_devices[index];`
```suggestion
          fprintf(stderr, "%s", "The device index provided is not correct. Will skip this device id.");
          continue;
```